src/main/java/co/inventorsoft/StreamHomework.java

```diff
@@ -29,17 +33,24 @@
      * @return collection of user, without duplicates
      */
     public List<User> createUsers(final List<String> emails) {
-        return null;
+        return emails.stream()
+        		.distinct()
+        		.filter(Objects::nonNull)
+        		.map(User::new)
+                .collect(Collectors.toList());
+
     }
 
+
     /**
      * Builds map with user email, as a key and user as a value.
      *
      * @param users collection of users
      * @return map {user email : user}
      */
     public Map<String, User> groupByEmail(final List<User> users) {
-        return null;
+    	return users.stream()
+                .collect(Collectors.toMap(User::getEmail, user->user));
```

Comment:
NTH: you could've used `Function.identity()` here. But this is also fine, there's no need to change it.